### PR TITLE
Add implementation for NSURL copy()

### PR DESCRIPTION
**Pull request by mbvreddy**
This PR has below implementations along with testcase

NSURL.copyWithZone()
NSURLQueryItem.copyWithZone()

### CODE REVIEW

File: TestFoundation/TestNSURL.swift [hunk 1]
```diff
@@ -417,6 +418,16 @@ class TestNSURL : XCTestCase {
             XCTAssertEqual(result, "file:///tmp/")
         }
     }
+
+    func test_copy() {
+        let url = NSURL(string: "https://www.swift.org")
+        let urlCopy = url!.copy() as! NSURL
+        XCTAssertTrue(url === urlCopy)
```
Comment (parkera) [inline TestFoundation/TestNSURL.swift]: This requires that the url instance is the same as the other; we may change that in the future. I would prefer the test checks a property of the URL for equality, or even better calls into URL's isEqual

File: TestFoundation/TestNSURL.swift [hunk 2]
```diff
@@ -417,6 +418,16 @@ class TestNSURL : XCTestCase {
...
+
+        let queryItem = NSURLQueryItem(name: "id", value: "23")
+        let queryItemCopy = queryItem.copy() as! NSURLQueryItem
+        XCTAssertTrue(queryItem === queryItemCopy)
```
Comment (parkera) [inline TestFoundation/TestNSURL.swift]: Same here

**Comment by mbvreddy** [comment]
@parkera I've squashed suggested changes. Please check.

**Comment by mbvreddy** [comment]
@parkera 

I've resolved conflicts.

NSURL.copy is already taken care by @phausler with commit https://github.com/apple/swift-corelibs-foundation/commit/6ce843275247e5fb1b42c66bb882cec0e3f7c8af

This PR now has implementation for NSURLQueryItem.copy() along with testcase.

**Comment by mbvreddy** [comment]
@parkera can this be merged?

**Comment by parkera** [comment]
Yes, this should be good.

**Comment by parkera** [comment]
@swift-ci please test

**Pull request merged**